app/controllers/users_controller.rb

```diff
@@ -8,23 +8,49 @@ def new
     end
   end
 
+  def confirm_email
+    user = User.find_by_confirm_token(params[:id])
+    if user
+      user.email_activate
+      session[:user_id] = user.id
+      redirect_to_role_portal(user)
+    end
+  end
+
   def create
     unless @user = existing_user(params[:user][:email])
       @user = User.new(user_params)
       @user.role = "parent"
     else
+      flash[:alert] = "This email is already in use, please use a different email"
       redirect_to signup_path
     end
 
     if @user.save
-      puts "#{@user.role} SAVED"
-      session[:user_id] = @user.id
-      redirect_to new_parent_parent_path
+      UserMailer.registration_confirmation(@user).deliver
+      flash[:notice] = "Thank you for creating an account. A confirmation email has been sent to the email you provided. Please click on the link to verify your account."
```

Comment:
Oh sure! haha I googled what to write as the confirmation message lol. I'll add it no prob!